test_util/azure_test_driver.py

```diff
@@ -1,226 +0,0 @@
-    '''
-    required = [
-        'AZURE_PARAM_agentEndpointDNSNamePrefix',
-        'AZURE_PARAM_linuxAdminUsername',
-        'AZURE_PARAM_masterEndpointDNSNamePrefix',
-        'AZURE_PARAM_sshRSAPublicKey',
-        'AZURE_TEMPLATE_URL'
-    ]
-    values = {}
-    for k in required:
-        assert os.getenv(k), "Environment variable {} must be set".format(k)
-        values[k] = os.getenv(k)
-
-    assert values['AZURE_PARAM_masterEndpointDNSNamePrefix'] != values['AZURE_PARAM_agentEndpointDNSNamePrefix'], \
-        "Master and agent prefix must be unique"
-
-
-def get_env_params():
-    '''
-    Return ARM input template parameters populated based on environment variables. Parameters are set as follows:
-      * AZURE_PARAM_key=value (value treated as a string)
-      * AZURE_PARAM_INT_key=value (value treated as an int)
-    '''
-    template_params = {}
-    for env_key, env_val in os.environ.items():
-        def extract_env_param(name, transform):
-            if env_key.startswith(name):
-                template_params[env_key[len(name):]] = {'value': transform(env_val)}
-                return True
-        if extract_env_param('AZURE_PARAM_INT_', int):
-            continue
-        extract_env_param('AZURE_PARAM_', lambda x: x)
-    return(template_params)
-
-
-def get_value(template_parameter):
-    return(get_env_params()[template_parameter]['value'])
-
-
-def get_test_config():
-    # Currently tests only pass with auth disabled, TEST_ADD_ENV_DCOS_AUTH_ENABLED=true
-    # can be set in the environment to override this
-    add_env = {'DCOS_AUTH_ENABLED': 'false'}
-    prefix = 'TEST_ADD_ENV_'
-    for k, v in os.environ.items():
-        if k.startswith(prefix):
-            add_env[k.replace(prefix, '')] = v
-    return add_env
-
-
-def main():
-    validate_env()
-    location = os.getenv('AZURE_LOCATION', 'East US')
-    credentials = azure.common.credentials.ServicePrincipalCredentials(
-        client_id=os.environ['AZURE_CLIENT_ID'],
-        secret=os.environ['AZURE_CLIENT_SECRET'],
-        tenant=os.environ['AZURE_TENANT_ID'])
-    subscription_id = os.environ['AZURE_SUBSCRIPTION_ID']
-    template = TemplateLink(uri=os.environ['AZURE_TEMPLATE_URL'])
-    # tenant_id = os.environ.get('AZURE_TENANT_ID')
-    # client_id = os.environ.get('AZURE_CLIENT_ID')
-    # client_secret = os.environ.get('AZURE_CLIENT_SECRET')
-    group_name = 'testing' + ''.join(random.choice('01234567890abcdef') for n in range(10))
-    deployment_name = 'deployment{}'.format(uuid.uuid4().hex)
-
-    rmc = ResourceManagementClient(credentials, subscription_id)
-
-    template_parameters = get_env_params()
-
-    # Output resource group
-    print("Resource group name: {}".format(group_name))
-    print("Deployment name: {}".format(deployment_name))
-
-    azure_cluster = {
-        'resource_group_name': group_name,
-        'deployment_name': deployment_name}
-    pkgpanda.util.write_json('azure-cluster.json', azure_cluster)
```

Comment:
This azure-cluster.json is necessary so that we can properly delete on error / have a two-stage job where if the first times out in TeamCity, the second will always attempt a delete